internal/storage/fs/snapshot.go

```diff
@@ -346,10 +353,80 @@ func (ss *storeSnapshot) addDoc(doc *ext.Document) error {
 				})
 			}
 
+			ruleRank += 1
+
 			ns.rules[rule.Id] = rule
 		}
 
 		ns.evalRules[f.Key] = evalRules
+
+		var rolloutRank int32 = 1
+		evalRollouts := make([]*storage.EvaluationRollout, 0, len(f.Rollouts))
+		for _, rollout := range f.Rollouts {
+			s := &storage.EvaluationRollout{
+				NamespaceKey: doc.Namespace,
+				Rank:         rolloutRank,
+			}
+
+			flagRollout := &flipt.Rollout{
+				Id:           uuid.Must(uuid.NewV4()).String(),
+				Rank:         rolloutRank,
+				FlagKey:      f.Key,
+				NamespaceKey: doc.Namespace,
+				CreatedAt:    ss.now,
+				UpdatedAt:    ss.now,
+			}
+
+			if rollout.Threshold != nil {
+				s.Threshold = &storage.RolloutThreshold{
+					Percentage: rollout.Threshold.Percentage,
+					Value:      rollout.Threshold.Value,
+				}
+				s.RolloutType = flipt.RolloutType_THRESHOLD_ROLLOUT_TYPE
+
+				flagRollout.Type = s.RolloutType
+				flagRollout.Rule = &flipt.Rollout_Threshold{
+					Threshold: &flipt.RolloutThreshold{
+						Percentage: rollout.Threshold.Percentage,
+						Value:      rollout.Threshold.Value,
+					},
+				}
+			} else if rollout.Segment != nil {
+				segment := ns.segments[rollout.Segment.Key]
```

Comment:
I think we should validate these lookups. Otherwise, this will cause a panic.
```suggestion
				segment, ok := ns.segments[rollout.Segment.Key]
				if !ok {
					return errs.ErrNotFoundf("segment %q not found", rollout.Segment.Key)
				}
```